ui/src/timeMachine/components/builderCard/BuilderCardDropdownHeader.tsx

```diff
@@ -20,17 +21,38 @@ export default class BuilderCardDropdownHeader extends PureComponent<Props> {
   }
 
   public render() {
-    const {children, options, onSelect, selectedOption, testID} = this.props
+    const {
+      children,
+      isInCheckOverlay,
+      options,
+      onSelect,
+      selectedOption,
+      testID,
+    } = this.props
+    const dropdownOptions = isInCheckOverlay ? ['filter'] : options
```

Comment:
this is better. the logic is now all in a single component and it's easier to follow.

However, it's mutating internal state needlessly. The length and content of options doesn't need to change - the options should always be `['filter', 'group']` - that represents what options are available to this dropdown. It's a dependency and it's communicating that it needs this dependency to function; we don't want to mutate that dependency. We can reduce this complexity and make the component conceptually simpler by not needing to worry about what the state of `options` is.

```suggestion

```